pkg/manager/tidbdashboard/manager.go

```diff
@@ -347,7 +347,9 @@ func EnsureImage(td *v1alpha1.TidbDashboard) string {
 		version := td.Spec.Version
 		if version == nil || *version == "" {
 			image = baseImage
-		} else if l := strings.LastIndex(baseImage, ":"); l >= 0 {
+		} else if l, s := strings.LastIndex(baseImage, ":"), strings.LastIndex(baseImage, "/"); l >= 0 && l > s {
+			// i.e. example.registry.com:30000/foo/pincap/tidb-dashboard:vx.x.x
+			// If there is only one ':', it MUST NOT be the seprator of hostname and port.
```

Comment:
I do not understand your comment here well. Do you mean for `example.registry.com:30000/foo/pincap/tidb-dashboard`, the whole part should be `baseImage` and no `verion` here, right? BTW, cloud you add some unit tests?

Reply:
Yes, it is what I mean. 
I will add some tests.

Reply:
@csuzhangxc  review again pls. 🥸